src/main/handlers/leaves.py

```diff
@@ -0,0 +1,70 @@
+        self.config = config
+
+        self.leaves_emoji = '🍃'
+        self.leaves_alias = ':leaves:'      
+        self.leaves_destroyer = ':fire:'
+        self.leaves_sources = ['🍃', '🌳']  
+        self.cooldown = 30
+
+        self.server_vars = {}
+
+    async def execute(self, message):
+        if not message.guild.id in self.server_vars:
+            self.server_vars[message.guild.id] = {}
+            self.server_vars[message.guild.id][ONGOING] = False
+            self.server_vars[message.guild.id][MESSAGE_COUNT] = 0
+            self.server_vars[message.guild.id][NUM_WAIT_MESSAGES] = 5
+            self.server_vars[message.guild.id][LEAVES_SENT] = 0
+            self.server_vars[message.guild.id][TIMES_REPEAT] = 5
+            self.server_vars[message.guild.id][LAST_LEAVES] = datetime.min
+
+        if self.server_vars[message.guild.id][ONGOING]:
+            if self.leaves_emoji in message.content:
+                self.server_vars[message.guild.id][MESSAGE_COUNT] = 0
+            else:
+                self.server_vars[message.guild.id][MESSAGE_COUNT] += 1
+
+                if self.server_vars[message.guild.id][MESSAGE_COUNT] == self.server_vars[message.guild.id][NUM_WAIT_MESSAGES]:
+                    if self.server_vars[message.guild.id][LEAVES_SENT] < self.server_vars[message.guild.id][TIMES_REPEAT]:
+                        await message.channel.send(content=self.leaves_alias)
+                        self.server_vars[message.guild.id][LEAVES_SENT] += 1
+                        self.server_vars[message.guild.id][NUM_WAIT_MESSAGES] = random.choice(range(3,6))
+                    elif self.server_vars[message.guild.id][LEAVES_SENT] == self.server_vars[message.guild.id][TIMES_REPEAT]:
+                        await message.channel.send(content=random.choices((self.leaves_destroyer, random.choice(self.config['GIFs'])), weights=[90, 10])[0])
+                        self.server_vars[message.guild.id][ONGOING] = False
+                    self.server_vars[message.guild.id][MESSAGE_COUNT] = 0
+
+        if any(leaves_source in message.content for leaves_source in self.leaves_sources):
+            print(f'{datetime.now().isoformat()} Leaves triggered!')
+            print(message.author, '-', message.content, message.jump_url)
+
+            if (datetime.now() - self.server_vars[message.guild.id][LAST_LEAVES]) < timedelta(minutes=self.cooldown):
+                print(f'{datetime.now().isoformat()} Leaves is on cooldown for '
+                    f'{self.cooldown - ((datetime.now() - self.server_vars[message.guild.id][LAST_LEAVES]).seconds // 60)} minutes')
+            elif self.server_vars[message.guild.id][ONGOING]:
+                print('Another leaves is currently still going!')
+            else:
+                print(f'{datetime.now().isoformat()} Throwing leaves!')
+                self.server_vars[message.guild.id][TIMES_REPEAT] = random.choices((random.choice(range(1, 10)), 30), weights=[95, 5])[0]
```

Comment:
what's the point of a random choice on a random choice?

Reply:
Could improve the readability here if you think it's justified. The internal `random.choice` chooses a number from 1 through 10 with equal weight and the external `random.choices` chooses between that number and a 5% chance of repeating a ridiculous number of times, 30 in this case.